agent/src/heapstats-engines/snapShotContainer.cpp

```diff
@@ -477,15 +477,19 @@ void TSnapShotContainer::mergeChildren(void) {
            * reference to it from child object data.
            */
           if (objData->isRemoved) {
+            TChildClassCounter *nextCounter = counter->next;
+
             if (prevCounter == NULL) {
-              srcClsCounter->child = counter->next;
+              srcClsCounter->child = nextCounter;
             } else {
-              prevCounter->next = counter->next;
+              prevCounter->next = nextCounter;
             }
 
             /* Deallocate TChildClassCounter. */
             free(counter->counter);
             free(counter);
+
+            counter = nextCounter;
```

Comment:
We set next child counter when current child counter is removed at this point.
This statement is not affected `free()`.